server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexStateService.java

```diff
@@ -1023,24 +1088,7 @@ public static ClusterBlock createUUIDBasedBlock(ClusterBlock clusterBlock) {
         );
     }
 
-    private static class OpenIndicesExecutor implements ClusterStateTaskExecutor<OpenIndicesTask> {
-
-        private final AllocationService allocationService;
-        private final IndexMetadataVerifier indexMetadataVerifier;
-        private final IndicesService indicesService;
-        private final ShardLimitValidator shardLimitValidator;
-
-        OpenIndicesExecutor(
-            AllocationService allocationService,
-            IndexMetadataVerifier indexMetadataVerifier,
-            IndicesService indicesService,
-            ShardLimitValidator shardLimitValidator
-        ) {
-            this.allocationService = allocationService;
-            this.indexMetadataVerifier = indexMetadataVerifier;
-            this.indicesService = indicesService;
-            this.shardLimitValidator = shardLimitValidator;
-        }
+    private class OpenIndicesExecutor implements ClusterStateTaskExecutor<OpenIndicesTask> {
```

Comment:
Slight preference to keep this (and other executors) as static classes that hold onto the services they need directly.

Reply:
Ehhhh, this is a bit of a pain -- `private class WaitForClosedBlocksApplied` / ` private class WaitForBlocksApplied`, and the former is invoked by `AddBlocksToCloseExecutor`, so I'd have to walk those two back, too.

I'm really with you on disliking (non-static) inner classes, they feel a bit too ['more magic'](http://www.catb.org/jargon/html/magic-story.html) for me, but in this case they're all private anyway, and the juggling of the fields across them all just seems like noise.

That said, this isn't 'no', it's just 'ehhh, this is ever so slightly harder than you might have thought' -- feel free to say something like "yeah, that's cool, please make the change anyway" and I will happily do it.

Reply:
My preference is only slight, I'll defer to your judgement.